packages/quantic/force-app/custom-token/main/default/classes/CustomTokenProvider.cls

```diff
@@ -0,0 +1,119 @@
+/**
+ * Example class showing off how to create your own TokenProvider,
+ * using a [CustomSettings](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_custom_settings.htm)
+ * to store the sensitive data such as the API Key.
+ */
+global with sharing class CustomTokenProvider implements ITokenProvider {
+    global with sharing class HttpException extends Exception {}
+    global with sharing class InvalidTokenSettingsException extends Exception {}
+
+    private static final String CONTENT_TYPE_APPLICATION_JSON = 'application/json; charset=UTF-8';
+    private static final String TOKEN_PATH = '/rest/search/v2/token';
+
+    private static Boolean isValidURL(String candidateURL) {
+        if(String.isEmpty(candidateURL)) {
```

Comment:
You can remove the `if` because `new Url()` would fail anyway if `candidateURL` is empty.